### DISCUSSION TITLE
Welcome to manim Discussions!

### DISCUSSION THREAD

**Discussion by 3b1b** [3b1b/manim#1302] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by TonyCrane** [comment]
Hi, everyone!

We just enabled the **[discussion](https://github.com/3b1b/manim/discussions) page**.  Because this is a more free discussion area than issues.

Here, you can ask questions you meet when using manim in the [**Q&A** category](https://github.com/3b1b/manim/discussions/categories/q-a).  
You can also share the amazing works you have made using manim and the good learning resources you find in [**show and tell**](https://github.com/3b1b/manim/discussions/categories/show-and-tell).  
In [**ideas**](https://github.com/3b1b/manim/discussions/categories/ideas), you can share the new features that you think manim can add, etc.  
Of course, you can also freely share anything you want to say in the [**general** category ](https://github.com/3b1b/manim/discussions/categories/general).

Hope that you can make reasonable use of this discussion community and enjoy yourself! :D

**Comment by gzhechu** [comment]
Hi, I am new to manimlib, and try to write some basic geometry animations. I don't know if it's OK to poste my idea here, so if it's not the right place, please let me know.
I noticed that manimlib uses only one CPU core to render and combine mp4 files. I don't know if there any guy tried use multi CPU core before.
So I did some modification on class Scene and Filewriter, add a "-j" parameter to specify how many job thread, and split all animations to several segement, and deliver to each thread, each thread has different 'start_at_animation_number' and 'end_at_animation_number', and then combine all the small pieces together.
I am using Ubuntu 20.04 with python 3.8, PIP installed manimlib. I met following problems, maybe some of you already know the issues before.
1. It seems each thread shared same 'ffmpeg' stdin pipe in same process, I'm not sure. It caused chaosed mp4 pieces.
2. So I turn to use numpy.save(each frame) to .npy files, and then finally combine to mp4 file, but meet '**cairo-path-fixed.c:839: _cairo_path_fixed_interpret: Assertion `!"reached"**' failed'. It seems cario is not thread safe, I'm still trying to figure out.
3. OpenCV cv2.VideoWriter seems another way to dump mp4 file , but I must solve the cario problem first.
4. Now my temprary solution is: wrote a python shell script to parse -j and start several REAL manim process(not thread), and then combine the final mp4 file. it probably practicable.

**Reply by TonyCrane** [reply]
- Firstly, you’d better open a new discussion [here](https://github.com/3b1b/manim/discussions/new), and choose the “**Ideas**“ category to post your idea.
- Secondly, the version here in the master branch is a stable version using CPU. If you want to improve it, you can go to [ManimCommunity/manim](https://github.com/ManimCommunity/manim), where the developers may be trying to achieve the same goal as you.
- Thirdly, the version here in the shaders branch is under development, which uses GPU to render, so you don’t need to worry about the efficiency.

**Reply by gzhechu** [reply]
Got it, thank you. I've clone CE version of manim, and started digging.

**Reply by mcurti** [reply]
- A similar question here. I use `ArrowVectorField` to generate a vector field from a class instance which uses a PDE toolbox to create the field. In `manim` scene I locate pointwise the points requested by the vector `ArrowVectorField`. Now the problem is that for larger resolution of arrows, this becomes extremely slow, so either I need to parallelize it somehow, or feed directly the arrays with values directly to `ArrowVectorField`. As far as I can see, at this moment I can use only a function that would accept the input `pos`
![MagnetField_ManimCE_v0 15 1](https://user-images.githubusercontent.com/31895988/162537656-f369c8a6-6ffe-484e-9b9e-09981a94a74c.png)
.

**Comment by HanokhUT** [comment]
Hello 3b1b, both my MikTex and Manimgl have been installed correctly, but when running your demonstration file, the error message "LaTeX compilation failed" still appears. How can I solve this problem?

**Reply by HanokhUT** [reply]
Traceback (most recent call last):d \\\quad \\\end{array}\right]...
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "D:\Mamba\envs\manimgl\Scripts\manimgl.exe\__main__.py", line 7, in <module>
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\__main__.py", line 61, in main
    run_scenes()
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\__main__.py", line 39, in run_scenes
    scene.run()
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\scene\scene.py", line 155, in run
    self.construct()
  File "D:\Program Files\manim-master\example_scenes.py", line 29, in construct
    IntegerMatrix(matrix),
    ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\mobject\matrix.py", line 245, in __init__
    super().__init__(matrix, num_decimal_places, decimal_config, **config)
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\mobject\matrix.py", line 224, in __init__
    super().__init__(
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\mobject\matrix.py", line 59, in __init__
    self.brackets = self.create_brackets(self.rows, bracket_v_buff, bracket_h_buff)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\mobject\matrix.py", line 120, in create_brackets
    brackets = Tex("".join((
               ^^^^^^^^^^^^^
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\mobject\svg\tex_mobject.py", line 59, in __init__
    super().__init__(
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\mobject\svg\string_mobject.py", line 69, in __init__
    svg_string = self.get_svg_string()
                 ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\mobject\svg\string_mobject.py", line 77, in get_svg_string
    return self.get_svg_string_by_content(content)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\mobject\svg\tex_mobject.py", line 70, in get_svg_string_by_content
    return latex_to_svg(content, self.template, self.additional_preamble, short_tex=self.tex_string)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\utils\tex_file_writing.py", line 84, in latex_to_svg
    return full_tex_to_svg(full_tex, compiler, message)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\utils\cache.py", line 27, in wrapper
    value = func(*args, **kwargs)
            ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Mamba\envs\manimgl\Lib\site-packages\manimlib\utils\tex_file_writing.py", line 130, in full_tex_to_svg
    raise LatexError(error_str or "LaTeX compilation failed")
manimlib.utils.tex_file_writing.LatexError: LaTeX compilation failed
